src/app/clusters/bindings/BindingManager.h

```diff
@@ -123,22 +121,46 @@ class BindingManager
     static BindingManager & GetInstance() { return sBindingManager; }
 
 private:
-    static BindingManager sBindingManager;
-
-    static void HandleDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr, SessionHandle & sessionHandle);
-    void HandleDeviceConnected(Messaging::ExchangeManager & exchangeMgr, SessionHandle & sessionHandle);
+    class ConnectionCallback
+    {
+    public:
+        ConnectionCallback(BindingManager * bindingManager) :
```

Comment:
This should never be null right? If so, let's make this a reference to make that clearer.